### PR TITLE
rest:dial: Support sipindialog hook

### PR DESCRIPTION
**Pull request by avoylenko**
I want to receive the sip indialog messages(INFO,  MESSAGE) when placing an outbound call from JBZ system using the `createCall` api. At this moment, I have failed to find inbox solution thats why I pushing this PR. Awaiting your comments.

### CODE REVIEW

File: lib/tasks/rest_dial.js [hunk 1]
```diff
@@ -142,6 +143,16 @@ class TaskRestDial extends Task {
         this.logger.error({err}, 'Rest:dial:_onAmdEvent - error calling actionHook');
       });
   }
+
+  _initSipIndialogRequestListener(cs, dlg) {
```
Comment (davehorton) [inline lib/tasks/rest_dial.js:147]: please change the name to `_initSipRequestWithinDialogHandler`

**Review by davehorton — CHANGES_REQUESTED** [review]

**Comment by davehorton** [comment]
also, please make a companion PR to verb-speciifications

**Comment by avoylenko** [comment]
> also, please make a companion PR to verb-speciifications

verb specification https://github.com/jambonz/verb-specifications/pull/34

**Comment by avoylenko** [comment]
Also the swagger specification will be updated if this PR suits the platform requirements.

**Review by davehorton — APPROVED** [review]

**Pull request merged**